src/data_base.cpp

```diff
@@ -323,6 +323,21 @@ code data_base::reorganize(const config::checkpoint& fork_point,
     return result ? error::success : error::operation_failed;
 }
 
+code data_base::confirm(const hash_digest& block_hash,
+                        const size_t height)
+{   
+    code ec;
+
+    if ((ec = verify_confirm(*blocks_, block_hash, height)))
+        return error::operation_failed;
+    
+    // confirm
+    if (!blocks_->index(block_hash, height, false))
+        return error::operation_failed;
```

Comment:
I should be confirming transactions too. I realise that now.

I thought we don't need to call update on the block to associate the transactions. This is because the block is already candidated, the transactions will be associated properly with the block before we try to confirm the block. Am I right on this?

Am I missing something else?

Reply:
New commit adding transaction confirmation for the block being confirmed. Test verifies the confirmation.